File: Nodal_Discontinuous_Galerkin_Methods_Hesthaven/nudg++/trunk/Src/Codes3D/dtscale3D.cpp
// dtscale3D.m
// function dtscale = dtscale3D; 
// 2007/06/11
//---------------------------------------------------------
#include "NDGLib_headers.h"
#include "NDG3D.h"


//---------------------------------------------------------
double NDG3D::dtscale3D() const
//---------------------------------------------------------
{
  // function dtscale = dtscale3D;
  // Purpose : Compute inscribed sphere diameter as characteristic
  //           for grid to choose timestep

  return 1.0 / (Fscale.max_val()*N*N);
}

File: Nodal_Discontinuous_Galerkin_Methods_Hesthaven/Codes1.1/Codes3D/dtscale3D.m
function dtscale = dtscale3D; 

% function dtscale = dtscale3D;
% Purpose : Compute ratio between volume and face Jacobian as characteristic
%           for grid to choose timestep

Globals3D;

dtscale = 1./(max(max(Fscale))*(N)*(N));
return;
